vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridTestScrollingOver100kLinesIT.java

```diff
@@ -46,14 +46,18 @@ public void toStringIsUsedForObjectSerialization() {
         // Checks that all header and body cells have text content
         List<TestBenchElement> allCellContents = grid
                 .$("vaadin-grid-cell-content").all();
+        List<TestBenchElement> headerSlots = grid.$("thead slot").all();
+        List<TestBenchElement> bodySlots = grid.$("tbody slot").all();
+        Assert.assertTrue(headerSlots.size() > 0);
+        Assert.assertTrue(bodySlots.size() > 0);
+
         allCellContents.forEach(vgcc -> {
-            String slotName = vgcc.getAttribute("slot")
-                    .replace("vaadin-grid-cell-content-", "");
-            if (Integer.parseInt(slotName) <= 180) {
+            TestBenchElement slot = vgcc.getPropertyElement("assignedSlot");
+            if (headerSlots.contains(slot) || bodySlots.contains(slot)) {
```

Comment:
This test relied the cell content elements in the light DOM belonging to body and header starting from an arbitrary index 180.